code/backend/batch/utilities/helpers/azure_computer_vision_client.py

```diff
@@ -0,0 +1,82 @@
+import logging
+from typing import List
+from urllib.parse import urljoin
+from azure.identity import DefaultAzureCredential, get_bearer_token_provider
+
+import requests
+from requests import Response
+
+from .env_helper import EnvHelper
+
+logger = logging.getLogger(__name__)
+
+
+class AzureComputerVisionClient:
+
+    __TOKEN_SCOPE = "https://cognitiveservices.azure.com/.default"
+    __VECTORIZE_IMAGE_PATH = "computervision/retrieval:vectorizeImage"
+    __RESPONSE_VECTOR_KEY = "vector"
+
+    def __init__(self, env_helper: EnvHelper) -> None:
+        self.host = env_helper.AZURE_COMPUTER_VISION_ENDPOINT
+        self.timeout = env_helper.AZURE_COMPUTER_VISION_TIMEOUT
+        self.key = env_helper.AZURE_COMPUTER_VISION_KEY
+        self.use_keys = env_helper.is_auth_type_keys()
+        self.api_version = env_helper.AZURE_COMPUTER_VISION_VECTORIZE_IMAGE_API_VERSION
+        self.model_version = (
+            env_helper.AZURE_COMPUTER_VISION_VECTORIZE_IMAGE_MODEL_VERSION
+        )
+
+    def vectorize_image(self, image_url: str) -> List[float]:
+        logger.info(f"Making call to computer vision to vectorize image: {image_url}")
+        response = self.__make_request(image_url)
+        self.__validate_response(response)
+
+        response_json = self.__get_json_body(response)
+        return self.__get_vectors(response_json)
+
+    def __make_request(self, image_url: str) -> Response:
+        try:
+            headers = {}
+            if self.use_keys:
+                headers["Ocp-Apim-Subscription-Key"] = self.key
+            else:
+                token_provider = get_bearer_token_provider(
+                    DefaultAzureCredential(), self.__TOKEN_SCOPE
+                )
+                headers["Authorization"] = "Bearer " + token_provider()
+
+            return requests.post(
+                url=urljoin(self.host, self.__VECTORIZE_IMAGE_PATH),
+                params={
+                    "api-version": self.api_version,
+                    "model-version": self.model_version,
+                },
+                json={"url": image_url},
+                headers=headers,
+                timeout=self.timeout,
+            )
+        except Exception as e:
+            raise Exception("Call to vectorize image failed") from e
```

Comment:
may be add the `image_url` in the exception to know exactly which image failed ?